### PR TITLE
Add tool to format with pretty-please

### PR DESCRIPTION
**Pull request by jdisanti**
## Motivation and Context
This PR adds a tool that formats a cargo workspace using the dtolnay/prettyplease library to replace rustfmt since rustfmt fails to format most of our generated code now.

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

**Review by ysaito1001 — APPROVED** [review]

**Comment by jdisanti** [comment]
Looks like prettyplease has a bug that the SDK triggers:
```
    Checking aws-sdk-transcribestreaming v0.0.0-local (/home/build/workspace/smithy-rs/aws/sdk/build/aws-sdk/sdk/transcribestreaming)
error: unnecessary trailing semicolon
  --> /home/build/workspace/smithy-rs/aws/sdk/build/aws-sdk/sdk/transcribestreaming/src/event_stream_serde.rs:17:9
   |
17 |         ; Ok(::aws_smithy_eventstream::frame::Message::new_from_parts(headers, payload))
   |         ^ help: remove this semicolon
   |
   = note: `-D redundant-semicolons` implied by `-D warnings`
```

**Comment by jdisanti** [comment]
I removed the cargo metadata workspace resolution from the tool since that was what was causing it to fail for the SDK (since it runs before the SDK copies over the runtime crates, and thus, the runtime crates don't exist when resolving the Cargo.toml files with `cargo metadata`). Now it just recursively formats all Rust files in a given directory.

**Comment by jdisanti** [comment]
Judging from the codegen diff, prettyplease removes all non-doc comments. Not sure if this will work for us.

**Review by hlbarber — APPROVED** [review]

File: codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/ServerCodegenVisitor.kt [hunk 1]
```diff
@@ -261,13 +261,14 @@ open class ServerCodegenVisitor(
...
-            "cargo fmt".runCommand(
+            "please-fmt".runCommand(
                 fileManifest.baseDir,
                 timeout = settings.codegenConfig.formatTimeoutSeconds.toLong(),
             )
         } catch (err: CommandFailed) {
             logger.info(
-                "[rust-server-codegen] Failed to run cargo fmt: [${service.id}]\n${err.output}",
+                "Failed to run please-fmt (be sure to install the tool with " +
+                    "`cargo install --path tools/ci-build/please-fmt`): [${service.id}]\n${err.output}",
```
Comment (david-perez) [inline codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/ServerCodegenVisitor.kt:271]:
```suggestion
                    "`cargo install --locked --path tools/ci-build/please-fmt`): [${service.id}]\n${err.output}",
```

**Review by david-perez — COMMENTED** [review]
Closes https://github.com/awslabs/smithy-rs/issues/1526.

This would be nice but we need it to preserve comments :(

File: tools/ci-build/please-fmt/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+authors = ["AWS Rust SDK Team <aws-sdk-rust@amazon.com>"]
+description = "Cargo frontend for the dtolnay/pretty-please library"
+edition = "2021"
+license = "Apache-2.0"
+publish = false
+
+[workspace]
+
+[dependencies]
+argh = "0.1"
```
Comment (david-perez) [inline tools/ci-build/please-fmt/Cargo.toml:13]: What's the benefit of this over `clap`?

**Comment by jdisanti** [comment]
Prettyplease doesn't preserve comments and [the maintainer doesn't want to support that](https://github.com/dtolnay/prettyplease/issues/50), so it's not a viable option for smithy-rs.

**Pull request closed without merging**